packages/analytics-plugin-google-global-site-tag/src/browser.ts

```diff
@@ -0,0 +1,528 @@
+  category: string,
+  value?: number,
+  nonInteraction?: boolean,
+  campaign?: ICampaignDataProps,
+}
+
+declare global {
+  interface Window {
+    dataLayer: unknown[];
+    gtag: ReturnType<typeof setUpWindowGtag>;
+  }
+}
+
+const defaultConfig = {
+  sendPageView: true,
+  customDimensions: {},
+  anonymizeIp: false,
+  allowGoogleSignals: true,
+  allowAdPersonalizationSignals: true,
+}
+
+let loadedInstances: { [key: string]: boolean } = {};
+
+const googleGtagAnalytics = (pluginConfig: IPluginConfig = {}) => {
+  let pageCalledOnce = false
+  // Allow for multiple google analytics instances
+  const instanceName = pluginConfig.instanceName ? pluginConfig.instanceName : '';
+
+  const trackingId = pluginConfig.trackingId;
+
+  return {
+    name: 'google-gtag-analytics',
+    config: {
+      ...defaultConfig,
+      ...pluginConfig
+    },
+    
+    // Load gtag.js and define gtag
+    // Set custom dimensions (ua properties) and parameters (ga4 properties)
+    initialize: (pluginApi: IPluginApiProps) => {
+      const { config, instance } = pluginApi;
+      if (!trackingId) throw new Error('No GA trackingId defined');
+
+      const gtagScriptSource = 'https://www.googletagmanager.com/gtag/js';
+      const scriptSrc = config.customScriptSrc || `${gtagScriptSource}?id=${trackingId}`;
+
+      if (!gtagLoaded(config.customScriptSrc || gtagScriptSource)) {
+        injectScript(scriptSrc);
+      }
+
+      if (!window.gtag) {
+        setUpWindowGtag();
+      }
+
+      /**
+       * Covert custom dimensions to:
+       * { dimension1: traitOne, dimension2: traitTwo }
+       */
+      const customDimensions = formatCustomDimensionsIntoCustomMap(config);
+
+      /**
+       * Convert `cookieConfig`'s keys from camel case to snake cases
+       */
+      let newCookieConfig: { [key: string]: unknown} = {};
+      if (config.cookieConfig !== undefined) {
+        Object.entries(config.cookieConfig).forEach(([key, value]) => {
+          newCookieConfig[camelToSnakeCase(key)] = value;
+        })
+      }
+
+      // Initialize tracker instance on page
+      if (!loadedInstances[instanceName]) {
+        let gtagConfig: { [key: string]: unknown } = {
+          cookie_domain: config.domain || 'auto',
+          send_page_view: config.sendPageView ? config.sendPageView : true,
+          allow_google_signals: config.allowGoogleSignals,
+          allow_ad_personalization_signals: config.allowAdPersonalizationSignals,
+          custom_map: customDimensions,
+          anonymize_ip: config.anonymizeIp,
+          ...newCookieConfig,
+        }
+
+        if (config.linker) {
+          gtagConfig.linker = config.linker;
+        }
+
+        /* set custom dimensions from user traits */
+        const user = instance.user() || {}
+        const traits = user.traits || {}
+
+        if (Object.keys(traits).length) {
+          window.gtag('set', traits);
+        }
+
+        window.gtag('config', trackingId, gtagConfig);
```

Comment:
I need to do more research but for the `gtagConfig` we passed to the `config` command, do we need to re-config these when we send a page view? Based on [this discussion on stack overflow](https://stackoverflow.com/questions/48120557/gtag-not-sending-custom-dimensions-for-events/48189202#48189202), it seems you have to pass `custom_map` every time with the config command with a page view update, but what about for other attributes like `send_page_view`?